### ISSUE TITLE
Parallel routes do not update metadata on link navigation

### ISSUE THREAD

**Issue by Fredkiss3** [vercel/next.js#48548]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Tue Apr 18 2023 16:54:41 GMT+0200 (heure d’été d’Europe centrale)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.19
      pnpm: 8.2.0
    Relevant packages:
      next: 13.3.0
      eslint-config-next: N/A
      react: N/A
      react-dom: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue

https://github.com/Fredkiss3/next-bug-parallel-routes-metadata

### To Reproduce

1. Open the project on stackblitz & start the project
2. open the preview in a new tab (otherwise it won't show)
3. From the home page click on any of the links, you will see that the title of the tab is still the one set on the homepage instead of the one set in `generateMetadata` of the parallel page
4. Reload the page while being on the parallel and the title of the tab will show the title set in `generateMetadata` of the parallel page

### Describe the Bug

While navigating with the link component from a normal to a parallel page, metadata are not applied, and only applied when doing a hard refresh while on the parallel page.

### Expected Behavior

The metadata should be applied when navigating like it is for normal page navigations

**Comment by sillvva** [comment]
A user in the [Discord](https://discord.gg/nextjs) said they resolved the issue by turning off Server Actions.

https://discord.com/channels/752553802359505017/1109962154259533875/1109967473635250310

I also noticed the issue when I started experimenting with Server Actions.

**Comment by Fredkiss3** [comment]
There is no server actions in this code but thanks anyway

**Comment by SnekCode** [comment]
@Fredkiss3 I ran into this issue today.  I was able to fix it by removing serverActions from my next.config.js
```
const nextConfig = {
    experimental: {
        serverActions: true, <--❌
    }
}
```

**Comment by Fredkiss3** [comment]
@sillvva  @SnekCode server actions are not activated in the code

**Comment by Fredkiss3** [comment]
this issue seems to have been fixed as of `next@13.4.10`
